File: TableGames.Games/PlayerCup.cs
using System.Collections.Generic;
using System.Linq;
using TableGames.Domain;

namespace TableGames.Games
{
    public class PlayerCup
    {
        public Player Player { get; set; }
        public List<Dice> Dices { get; set; }

        public PlayerCup(Player player) {
            Player = player;
            Dices = new List<Dice>();
        }

        public void ExposeDices() {
            Dices.ForEach(d => d.IsExposed = true);
        }

        public virtual object ToClient() {
            return new {
                playerName = Player.Name,
                dices = Dices.Select(d => d.ToClient()),
            };
        }
    }
}

File: TableGames.Games/playerCup.js
define(['knockout', 'game!dice'], function(ko, Dice) {

    function PlayerCup(playerCupState) {
        var self = this;
        self.playerName = playerCupState.playerName;
        self.dices = ko.observableArray(playerCupState.dices.map(function(diceState) { return new Dice(diceState); }));

        self.updateDices = function(dicesState) {
            dicesState.forEach(function(diceState, index) {
                var dice = self.dices()[index];
                if (dice) {
                    dice.value(diceState.value);
                    dice.isExposed(diceState.isExposed);
                    dice.isHighlighted(diceState.isHighlighted);
                } else {
                    self.dices.push(new Dice(diceState));
                }
            });
        };
    }

    return PlayerCup;
});
